paper

```diff
@@ -0,0 +1,298 @@
+          s|server) target_task=rebuildServerPatches;;
+          *) printf 'Unknown target - %s\nExpected one of: all api server\n' "${2:-}" >&2; return 1;;
+        esac
+        cd "${basedir}"
+        if [[ "${1}" == @(rbf|rbfull) ]]; then
+          printf 'Rebuilding patches without filter\n' >&2
+          ./gradlew "${target_task}" --filter-patches=false || return "${?}"
+        else
+          ./gradlew "${target_task}" || return "${?}"
+        fi
+      );;
+    p|patch)
+      (
+        local target_task
+        case "${2:-}" in
+          ''|all) target_task=applyPatches;;
+          a|api) target_task=applyApiPatches;;
+          s|server) target_task=applyServerPatches;;
+          *) printf 'Unknown target - %s\nExpected one of: all api server\n' "${2:-}" >&2; return 1;;
+        esac
+        cd "${basedir}"
+        ./gradlew "${target_task}" || return "${?}"
+      );;
+    j|jar)
+      (
+        cd "${basedir}"
+        ./gradlew applyPatches || return "${?}"
+        ./gradlew paperclipJar || return "${?}"
+      );;
+    b|build)
+      (
+      printf 'NOT IMPLEMENTED YET\n' >&2 # TODO
+      return 1
+      );;
+    i|install)
+      (
+        cd "${basedir}"
+        ./gradlew applyPatches || return "${?}"
+        ./gradlew publishToMavenLocal || return "${?}"
+        printf "Paper jar successfully built and installed to local repo\n"
+      );;
+    pc|paperclip)
+      (
+        cd "${basedir}"
+        ./gradlew paperclipJar || return "${?}"
+      );;
+    make)
+      (
+        if [[ "${2:-}" = bacon ]]; then
+          cd "${basedir}"
+          ./gradlew applyPatches || return "${?}"
+          ./gradlew paperclipJar || return "${?}"
+        fi
+      );;
+    m|mcdev)
+      (
+        cd "${basedir}"
+        ./gradlew makeMcDevSrc || return "${?}"
+      );;
+    t|test|testserver)
+      (
+        local target_task
+        case "${2:-}" in
+          ''|d|dev) target_task=runDev;;
+          s|shadow) target_task=runShadow;;
+          r|reobf) target_task=runReobf;;
+          *) printf 'Unknown target - %s\nExpected one of: dev shadow reobf\n' "${2:-}" >&2; return 1;;
+        esac
+        cd "${basedir}"
+        ./gradlew "${target_task}" || return "${?}"
+      );;
+    u|up|upstream)
+      (
+        cd "${basedir}"
+        ./scripts/upstreamMerge.sh "${2:-}" || return "${?}"
+      );;
+    cu|commitup|commitupstream|upc|upcommit|upstreamcommit)
+      (
+        cd "${basedir}"
+        shift
+        ./scripts/upstreamCommit.sh "${@}" || return "${?}"
+      );;
+    c|clean)
+      (
+        cd "${basedir}"
+        # Is this good or should be OG version?
+        ./gradlew clean cleanCache || return "${?}"
+      );;
+    con|continue)
+        if [ -d ".git/rebase-apply" ]; then
+            _papergit am --continue
+        elif [ -d ".git/rebase-merge" ]; then
+            _papergit rebase --continue
+        fi
+      ;;
+    e|edit)
```

Comment:
Would love if we could get rid of these entirely tbh.

These can cause issues esp when used by someone who doesn't know what they're doing. I think it's best to just not have it at all and have people do this themselves (it's not complicated).

But if there's pushback then sure, leave it.

Reply:
The way this command worked was always such a clusterfuck that if it disappeared, I really wouldn't give a single hoot, edit could maybe exist and just run the command for you, but, otherwise, the continue commands definitely were always a pain in the ass, you couldn't edit API and server at the same time and continue, and in general it was just kinda cludgy

Reply:
If the issue is that you can't edit server and api at the same time, I believe it can be fixed.

I can see the value in keeping this, but fine if you want to remove it.

Reply:
I'm not exactly for or against it, but, I mean, the way continue works means that you can't edit both at the same time as it forgets the last one, it's tripped so many people up over the years with how it works, I'd maybe keep the edit stuff just for laziness, but, am not really sure about `edit continue`

Reply:
What if we make it so you can do:

```sh
$ paper edit api
$ paper edit server
$ paper edit continue # Will continue in server
$ paper edit continue
No prior edit work to continue - try explicitly passing "api" or "server"?
$ paper edit continue api
```

Other case:

```sh
$ paper edit api
$ paper edit server
$ paper edit continue api
$ paper edit continue # Will continue server? Though I feel like API would be more intuitive?
```

I feel like it's not a very straightforward command either way, and to use it you need some understanding and decent CONTRIBUTING docs.

The more I look at it, the more confusing it becomes :smiley:

Reply:
That's the issue, the more you toy with it the more of a clusterfuck it becomes to the degree that I don't entirely see the merit in edit continue

Reply:
That CLI interface looks really confusing....I don't think this is benefiting anyone.

Reply:
Removed `edit continue`, left `edit api` and `edit server`.

Reply:
Why not do something like this: `paper edit api continue` and `paper edit server continue` (that's what [I did](https://github.com/Minebench/Origami/blob/1.16/origami#L56-L76) in my fork script), that way one can execute that extremely fast with the sequence `UP key` `SPACE` `c` `ENTER` while still being able to specify what to continue without manually doing it inside the folders.

Not having a continue function there would only resolve half of the complexity of the git actions, having a dedicated continue argument per type would fix the old issues with editing both at the same time as well as it sometimes not properly storing the edited type for the automatic continue.